packages/compat/src/keypair.ts

```diff
@@ -7,12 +7,8 @@ import { Keypair } from '@solana/web3.js';
  * @returns         A CryptoKeyPair
  */
 export async function fromLegacyKeypair(keypair: Keypair, extractable?: boolean): Promise<CryptoKeyPair> {
-    const [publicKey, privateKey] = await Promise.all([
-        crypto.subtle.importKey('raw', keypair.publicKey.toBytes(), 'Ed25519', true, ['verify']),
-        createPrivateKeyFromBytes(keypair.secretKey.slice(0, 32), extractable),
-    ]);
-    return {
-        privateKey,
-        publicKey,
-    } as CryptoKeyPair;
+    return createKeyPairFromBytes(
+        new Uint8Array([...keypair.secretKey.slice(0, 32), ...keypair.publicKey.toBytes()]),
```

Comment:
Technically you can just do this, since the `secretKey` is the private scalar (32 bytes) plus the public key (32 bytes).
```suggestion
        keypair.secretKey,
```